lib/src/facts/osx/operating_system_resolver.cc

```diff
@@ -33,8 +36,21 @@ namespace facter { namespace facts { namespace osx {
                 result.osx.version = move(value);
             }
 
+            // Populate the major and minor version facts using kernel_release
+            auto release = facts.get<string_value>(fact::kernel_release);
+
+            if (!release->value().empty()) {
+                result.release = move(release->value());
+                auto pos = release->value().find('.');
+                if (pos != string::npos) {
+                    auto second = release->value().find('.', pos + 1);
+                    result.major = release->value().substr(0, pos);
+                    result.minor = release->value().substr(pos + 1, second - (pos + 1));
```

Comment:
Can we guard against the unlikely event that OSX changes to just X.Y version numbers?